### PR TITLE
Take 2 at fixing dupes between refs and packages

### PR DESCRIPTION
**Pull request by davidfowl**
- Treat reference assemblies like assets of a node instead of
  dependencies of a node. They are to be treated like regular compile
  items in the project.lock.json and nothing more.
- Remove all of the special logic around special casing them during the
  graph walk.

Addresses https://github.com/NuGet/Home/issues/2332
/cc @emgarten

### CODE REVIEW

**Comment by analogrelay** [comment]
LGTM

**Comment by kevinchalet** [comment]
Merge, merge, merge! :open_mouth:

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreGraph.cs [hunk 1]
```diff
@@ -135,8 +135,7 @@ public static RestoreTargetGraph Create(bool writeToLockFile, IEnumerable<GraphN
 
                         if (string.Equals(node?.Item?.Key?.Type, LibraryTypes.Unresolved))
                         {
-                            if (node.Key.TypeConstraint != LibraryDependencyTarget.Reference
-                                && node.Key.VersionRange != null)
+                            if (node.Key.VersionRange != null)
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreGraph.cs:138]: Does this work if a version is defined by the framework reference? or is that no longer flowing through here?

**Comment by emgarten** [comment]
:shipit:

**Pull request closed without merging**